lib/tests/streamlit/streamlit_test.py

```diff
@@ -662,6 +663,24 @@ def test_st_title_with_anchor(self):
         self.assertEqual(el.heading.tag, "h1")
         self.assertEqual(el.heading.anchor, "some-anchor")
 
+    def test_st_time_input(self):
+        """Test st.time_input."""
+        value = datetime.time(8, 45)
+        st.time_input("Set an alarm for", value)
+
+        el = self.get_delta_from_queue().new_element
+        self.assertEqual(el.time_input.default, "08:45")
+        self.assertEqual(el.time_input.step, datetime.timedelta(minutes=15).seconds)
+
+    def test_st_time_input_with_step(self):
+        """Test st.time_input with step."""
+        value = datetime.time(9, 00)
+        st.time_input("Set an alarm for", value, step=datetime.timedelta(minutes=5))
```

Comment:
What will happen if I do `step=True` or `step="foo"` or `step=(90,)` ?

Reply:
I don't see any part of code that would actually check the type in runtime.

Reply:
In most of our components we don't do such checking and just rely on Type Hints and runtime exceptions, in my opinion it'd be a bit of over engineering, however I can add some type checking if you insist. But in such case I guess we need some approval for further messages send to user from product, since this is outside of the spec.

I'm leaving this as is for now and waiting for a response from you

Reply:
I'd say that all I'm asking boils down to changing your code from:

```
        if isinstance(step, timedelta):
            step = step.seconds
        if step < 60 or step > timedelta(hours=23).seconds:
            raise StreamlitAPIException(
                f"`step` must be between 60 seconds and 23 hours but is currently set to {step} seconds."
            )
```

to:

```
        if isinstance(step, timedelta):
            step = step.seconds
        elif not isinstance(step, int):
            raise StreamlitAPIException(f"`step` can only be `int` or `timedelta` but {type(step)} is provided")
        if step < 60 or step > timedelta(hours=23).seconds:
            raise StreamlitAPIException(
                f"`step` must be between 60 seconds and 23 hours but is currently set to {step} seconds."
            )
```

So we are talking about adding two lines of code :) IMO pretty cheap stuff to check, even if not done elsewhere.

Reply:
Yes, it's cheap and as I wrote I'm not going to insist on that, thank you for the example and the message, I've updated it with a small change

```
        if not isinstance(step, (int, timedelta)):
            raise StreamlitAPIException(
                f"`step` can only be `int` or `timedelta` but {type(step)} is provided."
            )
```

this is called before any `step` conversion